app/models/manageiq/providers/base_manager/refresher.rb

```diff
@@ -165,8 +165,8 @@ def parse_targeted_inventory(ems, target, collector)
           log_header = format_ems_for_logging(ems)
           _log.debug("#{log_header} Parsing inventory...")
           persister, = Benchmark.realtime_block(:parse_inventory) do
-            persister = ManageIQ::Providers::Inventory.persister_class_for(target.class).new(ems, target)
-            parser = ManageIQ::Providers::Inventory.parser_class_for(target.class).new
+            persister = ManageIQ::Providers::Inventory.persister_class_for(ems, target, target.class.name.demodulize).new(ems, target)
+            parser = ManageIQ::Providers::Inventory.parser_class_for(ems, target, target.class.name.demodulize).new
```

Comment:
adding `target.class.name.demodulize` makes the same behaviour as old `parser_class_for` method. 
I choose this solution rather than some `old_parser_class_for` method
Problem is that Amazon's Storage manager's parser and AnsibleTower::AutomationManager::ConfigurationScriptSource's parser doesn't have compatible "suffix"